egs/wsj/s5/utils/data/get_num_frames.sh

```diff
@@ -22,4 +22,4 @@ fi
 
 frame_shift=$(utils/data/get_frame_shift.sh $data) || exit 1
 
-awk -v s=$frame_shift '{n += $2} END{printf("%d\n", int(n / s))}' <$data/utt2dur
+python utils/data/get_num_frames.py --frame-shift $frame_shift --utt2dur-dir $data/utt2dur
```

Comment:
Hm.  So one line of awk transforms into 24 lines of python.
Did you try the fix of just changing ` n += $2` to `n += 1.0 * $2` so it uses floating-point, not integer arithmetic?
The result doesn't have to be exact.